include/deal.II/numerics/vector_tools.h

```diff
@@ -1858,41 +1900,41 @@ namespace VectorTools
    *
    * See the general documentation of this namespace for further information.
    */
-  template <int dim, int spacedim>
+  template <int dim, typename VectorType, int spacedim>
   void create_right_hand_side (const Mapping<dim, spacedim>    &mapping,
                                const DoFHandler<dim,spacedim>  &dof,
                                const Quadrature<dim>           &q,
                                const Function<spacedim,double> &rhs,
-                               Vector<double>                  &rhs_vector);
+                               VectorType                      &rhs_vector);
```

Comment:
I don't really have a preference, but at least `VectorTools::point_difference`, `VectorTools::point_value`, `VectorTools::point_gradient` and `VectorTools::compute_mean_value` use the same format. Since the other `VectorTools::project_*` use the format you are proposing, I'll follow your suggestion.